### PR TITLE
refac: add monorepo structure with fuels default packages config

### PR DESCRIPTION
**Pull request by pedronauck**
This pull request is intended to make this application conform to the standard we use in our other repositories.

## Main Changes

- [x] Use PNPM with Turborepo and workspaces, instead of Yarn
- [x] App is inside `packages/app` folder instead of `client/` 
- [x] Use ViteJS instead of Create React App
- [x] Add all our default configurations for tooling (eslint, jest, prettier, etc)

✨ Now we will use `pnpm dev` to run the application instead of `yarn start`

### CODE REVIEW

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: client/.env.example [hunk 1]
```diff
@@ -1,5 +0,0 @@
-REACT_APP_FUEL_PROVIDER_URL=https://node.swayswap.io/graphql
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply client/.env.example:1]: This was just moved to the new folder, its side `packages/app/.env.example`

File: .prettierrc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "overrides": [
     {
-      "files": ["*.js", "*.ts", ".tsx"],
+      "files": ["*.js", "*.ts"],
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply .prettierrc:4]: ✅

File: contracts/token_contract/.rustc_info.json [hunk 3]
```diff
@@ -1 +1,34 @@
-{"rustc_fingerprint":4025667977361673173,"outputs":{"931469667778813386":{"success":true,"status":"","code":0,"stdout":"___\nlib___.rlib\nlib___.dylib\nlib___.dylib\nlib___.a\nlib___.dylib\n/Users/mohammad/.rustup/toolchains/stable-aarch64-apple-darwin\ndebug_assertions\nproc_macro\ntarget_arch=\"aarch64\"\ntarget_endian=\"little\"\ntarget_env=\"\"\ntarget_family=\"unix\"\ntarget_os=\"macos\"\ntarget_pointer_width=\"64\"\ntarget_vendor=\"apple\"\nunix\n","stderr":""},"2797684049618456168":{"success":true,"status":"","code":0,"stdout":"___\nlib___.rlib\nlib___.dylib\nlib___.dylib\nlib___.a\nlib___.dylib\n","stderr":""},"5309432699494263626":{"success":true,"status":"","code":0,"stdout":"___\nlib___.rlib\nlib___.dylib\nlib___.dylib\nlib___.a\nlib___.dylib\n","stderr":""},"17598535894874457435":{"success":true,"status":"","code":0,"stdout":"rustc 1.59.0 (9d1b2106e 2022-02-23)\nbinary: rustc\ncommit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a\ncommit-date: 2022-02-23\nhost: aarch64-apple-darwin\nrelease: 1.59.0\nLLVM version: 13.0.0\n","stderr":""}},"successes":{}}
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply contracts/token_contract/.rustc_info.json:1]: ✅

File: packages/app/deploy-contracts/index.ts [hunk 4]
```diff
@@ -25,8 +26,9 @@ const contractPath = path.join(
   __dirname,
   '../../contracts/exchange_contract/out/debug/exchange_contract.bin'
 );
-const providerUrl = process.env.REACT_APP_FUEL_PROVIDER_URL || 'https://node.swayswap.io/graphql';
+const providerUrl = import.meta.env.VITE_FUEL_PROVIDER_URL || 'https://node.swayswap.io/graphql';
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/deploy-contracts/index.ts:29]: ✅

File: packages/app/src/components/CoinInput.tsx [hunk 5]
```diff
@@ -79,7 +80,9 @@ export function useCoinInput({
 }: UseCoinParams) {
   const [amount, setAmount] = useState<bigint | null>(null);
   const { data: balances } = useBalances({ enabled: showBalance });
-  const coinBalance = balances?.find((item) => item.assetId === coin?.assetId);
+  const coinBalance = balances?.find(
+    (item: any) => item.assetId === coin?.assetId
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply packages/app/src/components/CoinInput.tsx:84]: ✅

**Review by luizstacio — APPROVED** [review]
Nice...

**Pull request merged**